src/main/scala/io/iohk/ethereum/vm/Program.scala

```diff
@@ -16,6 +18,30 @@ case class Program(code: ByteString) {
   def getBytes(from: Int, size: Int): ByteString =
     code.slice(from, from + size).padTo(size, 0.toByte)
 
+  lazy val validJumpDestinations: Seq[Int] = validJumpDestinationsAfterPosition(0)
```

Comment:
Yes, thats why a precomputed set is used :+1:
I will replace the `validJumpDestinations` with a `Set[Int]`